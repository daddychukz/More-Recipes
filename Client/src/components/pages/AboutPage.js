import React, { Component } from 'react';

class AboutPage extends Component {
    render() {
        return (
            <div>
                <h1>About</h1>
                <p>ksajfvdkjv akvjaskdfc, kasfjsdf</p>
            </div>
        );
    }
}

export default AboutPage;