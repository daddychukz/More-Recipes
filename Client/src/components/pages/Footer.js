import React from 'react';
import { Link } from 'react-router-dom';

//Stateless component
export const Footer = () => {
        return(
            // MAIN FOOTER
            <footer id="main-footer" className="bg-dark">
                <div className="container">
                    <div className="row">
                        <div className="col text-center">
                            <div className="py-4">
                                <h1 className="h3">More-Recipes</h1>
                                <p>Copyright &copy; 2017</p>
                                {/* <button className="btn btn-primary" data-toggle="modal" data-target="#contactModal">Contact Us</button> */}
                            </div>
                        </div>
                    </div>
                </div>
            </footer>
        );
    }