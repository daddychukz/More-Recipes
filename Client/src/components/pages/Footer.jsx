import React from 'react';

// Stateless component
const Footer = () => (

  <footer className="bg-dark">
    <div className="py-4 txt">
      <h1 className="h3">More-Recipes</h1>
      <p>Copyright &copy; 2017</p>
    </div>
  </footer>

);

export default Footer;
