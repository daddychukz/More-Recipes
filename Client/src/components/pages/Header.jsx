import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import customHistory from '../common/commonFunctions';
import { logout } from '../../actions/userActions';

/**
 * @class Header
 *
 * @extends {React.Component}
 */
class Header extends React.Component {
  /**
   * @param {any} props
   *
   * @memberof Header
   */
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  /**
   * @description logs out a signed in user
   *
   * @memberof Header
   *
   * @returns {void}
   */
  logout() {
    this.props.logout();
    customHistory.push('/');
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof Header
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const { pathname } = window.location;

    const className = link => classNames({
      active: pathname === link,
      'nav-link': true
    });

    return (
      <div>
        <nav className="navbar navbar-expand-md bg-dark navbar-dark fixed-top">
          <div className="container">
            <Link to={'/recipe-box'} className="navbar-brand">
              <h3 id="logo">More-Recipes</h3>
            </Link>
            <button
              className="navbar-toggler"
              data-toggle="collapse"
              data-target="#navbarCollapse"
            >
              <span className="navbar-toggler-icon" />
            </button>
            <div className="collapse navbar-collapse" id="navbarCollapse">
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link
                    to={'/recipe-box'}
                    className={`nav-link ${pathname === '/recipe-box' &&
                    'active'}`}
                  >Home</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/add-recipe'} className={className('/add-recipe')}>
                  Add Recipe
                  </Link>
                </li>
                <li className="nav-item">
                  <Link to={'/my-recipe'} className={className('/my-recipe')}>
                  My Recipes
                  </Link>
                </li>
                <li className="nav-item">
                  <Link to={'/my-favorite'} className={className('/my-favorite')}>
                  Favorites
                  </Link>
                </li>
                <li className="nav-item">
                  <div className="btn-group open">
                    <i
                      className="fa fa-user-circle fa-2x pull-right"
                      data-toggle="dropdown"
                      aria-hidden="true"
                    />
                    <ul className="dropdown-menu">
                      <li>
                        <Link className="dropdown-item" to={'/my-profile'}>
                        Profile
                        </Link>
                      </li>
                      <li className="divider" />
                      <li>
                        <Link
                          className="dropdown-item"
                          to={'/'}
                          onClick={this.logout}
                        >
                          Logout
                        </Link>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </nav>

        <div className="modal fade text-dark" id="contactModal">
          <div className="modal-dialog">
            <div className="modal-content">
              <div className="modal-header bg-primary">
                <h5
                  className="modal-title"
                  style={{ color: 'white' }}
                  id="contactModalTitle"
                >
                  Contact Us
                </h5>
              </div>
              <div className="modal-body">
                <form>
                  <div className="form-group">
                    <label htmlFor="name">Name</label>
                    <input type="text" className="form-control" />
                  </div>
                  <div className="form-group">
                    <label htmlFor="email">Email</label>
                    <input type="email" className="form-control" />
                  </div>
                  <div className="form-group">
                    <label htmlFor="messager">Message</label>
                    <textarea className="form-control" />
                  </div>
                </form>
              </div>
              <div className="modal-footer">
                <button className="btn btn-primary btn-block">Submit</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


Header.propTypes = {
  logout: PropTypes.func.isRequired
};

export default connect(null, { logout })(Header);
