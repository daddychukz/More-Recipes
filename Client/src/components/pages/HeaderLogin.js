import React from 'react';
import { Link } from 'react-router-dom';

//Stateless component
export const Header = () => {
        return(
            <div>
                <nav className="navbar navbar-expand-sm bg-dark navbar-dark fixed-top">
                    <div className="container">
                        <Link to={'/'} className="navbar-brand">More-Recipes</Link>
                        <button className="navbar-toggler" data-toggle="collapse" data-target="#navbarCollapse">
                            <span className="navbar-toggler-icon"></span>
                        </button>
                        <div className="collapse navbar-collapse" id="navbarCollapse">
                            <ul className="navbar-nav ml-auto">
                                <li className="nav-item">
                                    <Link to={'/'}  className="nav-link active">Home</Link>
                                </li>
                                <li className="nav-item">
                                    <Link to="#" className="nav-link" data-toggle="modal" data-target="#contactModal">Contact</Link>
                                </li>
                            </ul>
                        </div>
                    </div>
                </nav>
                
                <div className="modal fade text-dark" id="contactModal">
                    <div className="modal-dialog">
                        <div className="modal-content">
                            <div className="modal-header bg-primary">
                                <h5 className="modal-title" style={{color:'white'}} id="contactModalTitle">
                                    Contact Us
                                </h5>
                            </div>
                            <div className="modal-body">
                                <form>
                                    <div className="form-group">
                                        <label htmlFor="name">Name</label>
                                        <input type="text" className="form-control" />
                                    </div>
                                    <div className="form-group">
                                        <label htmlFor="email">Email</label>
                                        <input type="email" className="form-control" />
                                    </div>
                                    <div className="form-group">
                                        <label htmlFor="messager">Message</label>
                                        <textarea className="form-control"></textarea>
                                    </div>
                                </form>
                            </div>
                            <div className="modal-footer">
                                <button className="btn btn-primary btn-block">Submit</button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            // <nav className="navbar navbar-expand-md bg-dark navbar-dark fixed-top">
            //     <div className="container">
            //         <Link to={'/'} className="navbar-brand"><h1 id="logo">More-Recipes</h1></Link>
            //         <button className="navbar-toggler" data-toggle="collapse" data-target="#navbarCollapse">
            //             <span className="navbar-toggler-icon"></span>
            //         </button>
            //         <div className="collapse navbar-collapse" id="navbarCollapse">
            //             <ul className="navbar-nav ml-auto">
            //                 <li className="nav-item">
            //                     <Link to={'/'}  className="nav-link active">Home</Link>
            //                 </li>
            //                 <li className="nav-item">
            //                     <Link to={'/add-recipe'} className="nav-link">Add Recipe</Link>
            //                 </li>
            //                 <li className="nav-item">
            //                     <Link to={'/my-recipe'} className="nav-link">My Recipes</Link>
            //                 </li>
            //                 <li className="nav-item">
            //                     <Link to={'/my-favorite'} className="nav-link">Favorites</Link>
            //                 </li>
            //                 <li className="nav-item">
            //                     <div className="btn-group open">
            //                         <i className="fa fa-user-circle fa-2x pull-right" data-toggle="dropdown" aria-hidden="true"></i>
            //                         <ul className="dropdown-menu">
            //                             <li><Link className="dropdown-item" to={'/profile'}>Profile</Link></li>
            //                             <li className="divider"></li>
            //                             <li><Link className="dropdown-item" to={'/login'}>Logout</Link></li>                        
            //                         </ul>
            //                     </div>                                                        
            //                 </li>
            //             </ul>
            //         </div>
            //     </div>
            // </nav>
        );
    }
