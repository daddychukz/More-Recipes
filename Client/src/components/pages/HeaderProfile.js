import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import createBrowserHistory from 'history/createBrowserHistory';
import PropTypes from 'prop-types';
import { logout } from '../../actions/userActions';

const customHistory = createBrowserHistory({
  forceRefresh: true
});

/**
 * 
 * 
 * @class HeaderProfile
 * @extends {React.Component}
 */
class HeaderProfile extends React.Component {
  /**
   * 
   * @returns {void}
   * @param {any} e 
   * @memberof HeaderProfile
   */
  logout(e) {
    this.props.logout();
    customHistory.push('/');
  }
  /**
   * 
   * 
   * @returns {object} component
   * @memberof HeaderProfile
   */
  render() {
    const { pathname } = window.location;

    const className = (link) => classNames({
      'active': pathname === link,
      'nav-link': true
    });

    return (
      <div>
        <nav className="navbar navbar-expand-md bg-dark navbar-dark fixed-top">
          <div className="container">
            <Link to={'/recipe-box'} className="navbar-brand"><h1 id="logo">More-Recipes</h1></Link>
            <button className="navbar-toggler" data-toggle="collapse" data-target="#navbarCollapse">
              <span className="navbar-toggler-icon" />
            </button>
            <div className="collapse navbar-collapse" id="navbarCollapse">
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link to={'/recipe-box'} className={`nav-link ${pathname === '/recipe-box' && 'active'}`}>Home</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/add-recipe'} className={className('/add-recipe')}>Add Recipe</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/my-recipe'} className={className('/my-recipe')}>My Recipes</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/my-favorite'} className={className('/my-favorite')}>Favorites</Link>
                </li>
                <li className="nav-item">
                  <div className="btn-group open">
                    <i className="fa fa-user-circle fa-2x pull-right" data-toggle="dropdown" aria-hidden="true" />
                    <ul className="dropdown-menu">
                      <li><Link className="dropdown-item" to={'/my-profile'}>Profile</Link></li>
                      <li className="divider" />
                      <li><Link className="dropdown-item" to="#" data-toggle="modal" data-target="#profileModal">Edit Profile</Link></li>
                      <li><Link className="dropdown-item" to={'/'} onClick={this.logout.bind(this)}>Logout</Link></li>
                    </ul>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    );
  }
}


HeaderProfile.propTypes = {
  logout: PropTypes.func.isRequired
};

export default connect(null, { logout })(HeaderProfile);
