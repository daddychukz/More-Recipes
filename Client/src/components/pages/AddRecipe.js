import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header';
import SideBar from './SideBar';
import { Footer } from './Footer';
import RecipeForm from '../forms/RecipeForm';

class AddRecipe extends React.Component {
    render() {
        return (
            <div>
                <Header />

                <header id="header" />

                {/* BREADCRUMB  */}
                <div className="container">
                    <nav className="breadcrumb">
                        <Link className="breadcrumb-item" to="#">Home</Link>
                        <span className="breadcrumb-item active">Recipe</span>
                    </nav>
                </div>

                {/* MAIN SECTION  */}
                <section>
                    <div className="container">
                        <div className="row">
                            <SideBar />
                            <div className="col-md-8">
                                <RecipeForm />
                            </div>
                        </div>
                    </div>
                </section>

                <Footer />

                {/* Recipe MODAL  */}
                <div className="modal fade text-dark" id="addRecipe">
                    <div className="modal-dialog">
                        <div className="modal-content">
                            <div className="modal-header bg-primary">
                                <h5 className="modal-title" style={{ color: 'rgb(255, 255, 255)' }} id="contactModalTitle">
                                            Add Recipe
                                </h5>
                            </div>
                            <div className="modal-body">
                                <form>
                                    <div className="form-group">
                                        <label htmlFor="name">Recipe Title</label>
                                        <input type="text" className="form-control" />
                                    </div>
                                    <div className="form-group">
                                        <label htmlFor="messager">Recipe Description</label>
                                        <textarea className="form-control" rows="6" />
                                    </div>
                                    <div className="form-group">
                                        <label htmlFor="imgFile">Upload Image</label>
                                        <input type="file" className="form-control-file" id="imgFile" aria-describedby="fileHelp" />
                                        <small id="fileHelp" className="form-text text-muted">Please upload an example image of recipe for better reviews.</small>
                                    </div>
                                </form>
                            </div>
                            <div className="modal-footer">
                                <button className="btn btn-primary btn-block" data-dismiss="modal">Submit</button>
                            </div>
                        </div>
                    </div>
                </div>

                {/* FAVORITE MODAL  */}
                <div className="modal fade text-dark" id="favorite" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
                    <div className="modal-dialog" role="document">
                        <div className="modal-content">
                            <div className="modal-header">
                                <h5 className="modal-title" id="exampleModalLabel">Message</h5>
                                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                    <span aria-hidden="true">&times;</span>
                                </button>
                            </div>
                            <div className="modal-body">
                                    Successfully Added to Your Favorites
                            </div>
                            <div className="modal-footer">
                                <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
                                <button type="button" className="btn btn-primary">Save</button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default AddRecipe;
