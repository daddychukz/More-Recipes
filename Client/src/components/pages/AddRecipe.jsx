import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header';
import SideBar from './SideBar';
import Footer from './Footer';
import RecipeForm from '../forms/RecipeForm';

const AddRecipe = () => (

  <div>
    <Header />

    <header id="header" />

    {/* BREADCRUMB  */}
    <div className="container">
      <nav className="breadcrumb">
        <Link className="breadcrumb-item" to="#">Home</Link>
        <span className="breadcrumb-item active">Recipe</span>
      </nav>
    </div>

    {/* MAIN SECTION  */}
    <section>
      <div className="container">
        <div className="row">
          <SideBar />
          <div className="col-md-12 col-lg-8">
            <RecipeForm />
          </div>
        </div>
      </div>
    </section>
    <br />

    <Footer />
  </div>
);

export default AddRecipe;
