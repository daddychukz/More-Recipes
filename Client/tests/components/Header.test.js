import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { PureHeader } from '../../src/components/pages/Header';


const props = {
  logout: sinon.spy(() => new Promise((cb) => {
    cb();
  }))
};

const shallowWrapper = shallow(<PureHeader {...props} />);

describe('MyFavorite test suite', () => {
  it('renders MyFavorite component without crashing', () => {
    expect(shallowWrapper).toBeDefined();
    expect(shallowWrapper.find('nav').length).toBe(1);
  });

  it('should match component snapshot', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('calls logout event', () => {
    sinon.spy(shallowWrapper.instance(), 'logout');
    shallowWrapper.instance().logout();
    expect(shallowWrapper.instance().logout.calledOnce).toEqual(true);
  });
});
