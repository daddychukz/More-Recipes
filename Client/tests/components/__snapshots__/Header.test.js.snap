// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFavorite test suite should match component snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md bg-dark navbar-dark fixed-top"
  >
    <div
      className="container"
    >
      <Link
        className="navbar-brand"
        replace={false}
        to="/recipe-box"
      >
        <h3
          id="logo"
        >
          More-Recipes
        </h3>
      </Link>
      <button
        className="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <Link
              className="nav-link false"
              id="home"
              replace={false}
              to="/recipe-box"
            >
              Home
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              id="addRecipe"
              replace={false}
              to="/add-recipe"
            >
              Add Recipe
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              id="my-recipe"
              replace={false}
              to="/my-recipe"
            >
              My Recipes
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              replace={false}
              to="/my-favorite"
            >
              Favorites
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <div
              className="btn-group open"
            >
              <i
                aria-hidden="true"
                className="fa fa-user-circle fa-2x pull-right"
                data-toggle="dropdown"
                id="drop-down"
              />
              <ul
                className="dropdown-menu"
              >
                <li>
                  <Link
                    className="dropdown-item"
                    replace={false}
                    to="/my-profile"
                  >
                    Profile
                  </Link>
                </li>
                <li
                  className="divider"
                />
                <li>
                  <Link
                    className="dropdown-item"
                    id="logout"
                    onClick={[Function]}
                    replace={false}
                    to="/"
                  >
                    Logout
                  </Link>
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="modal fade text-dark"
    id="contactModal"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header bg-primary"
        >
          <h5
            className="modal-title"
            id="contactModalTitle"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Contact Us
          </h5>
        </div>
        <div
          className="modal-body"
        >
          <form>
            <div
              className="form-group"
            >
              <label
                htmlFor="name"
              >
                Name
              </label>
              <input
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="form-control"
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="messager"
              >
                Message
              </label>
              <textarea
                className="form-control"
              />
            </div>
          </form>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary btn-block"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
